src/storage/src/source/mysql/statistics.rs

```diff
@@ -0,0 +1,179 @@
+) -> (
+    Stream<G, ProgressStatisticsUpdate>,
+    Stream<G, ReplicationError>,
+    PressOnDropButton,
+) {
+    let op_name = format!("MySqlStatistics({})", config.id);
+    let mut builder = AsyncOperatorBuilder::new(op_name, scope);
+
+    let (mut stats_output, stats_stream) = builder.new_output();
+
+    // TODO: Add additional metrics
+
+    let (button, transient_errors) = builder.build_fallible::<TransientError, _>(move |caps| {
+        Box::pin(async move {
+            let worker_id = config.worker_id;
+            let [stats_cap]: &mut [_; 1] = caps.try_into().unwrap();
+
+            // Only run the replication reader on the worker responsible for it.
+            if !config.responsible_for(STATISTICS) {
+                return Ok(());
+            }
+
+            let connection_config = connection
+                .connection
+                .config(
+                    &*config.config.connection_context.secrets_reader,
+                    &config.config,
+                )
+                .await?;
+
+            let mut stats_conn = connection_config
+                .connect(
+                    &format!("timely-{worker_id} MySQL replication statistics"),
+                    &config.config.connection_context.ssh_tunnel_manager,
+                )
+                .await?;
+
+            tokio::pin!(resume_uppers);
+
+            let mut offset_known = None;
+            let mut offset_committed = None;
+
+            let upstream_stream = async_stream::stream!({
+                // TODO(guswynn): make configurable (at the same time as the pg one).
+                let mut interval = tokio::time::interval(std::time::Duration::from_secs(10));
+                loop {
+                    interval.tick().await;
+
+                    let gtid_executed =
+                        query_sys_var(&mut stats_conn, "global.gtid_executed").await?;
+                    match gtid_set_frontier(&gtid_executed) {
+                        Ok(frontier) => yield Ok(frontier),
+                        Err(err) => {
+                            // We don't translate this into a definite error like in snapshotting, but we will
+                            // restart the source.
+                            yield Err::<_, TransientError>(err.into())
+                        }
+                    }
+                }
+            });
+            tokio::pin!(upstream_stream);
+
+            loop {
+                tokio::select! {
+                    Some(upstream_frontier) = upstream_stream.next() => {
+                        offset_known = Some(aggregate_mysql_frontier(upstream_frontier?));
+
+                    },
+                    Some(committed_frontier) = resume_uppers.next() => {
+                        offset_committed = Some(aggregate_mysql_frontier(committed_frontier));
+                    },
+                    else => break
+                };
+            }
```

Comment:
won't this just loop forever? or do those streams yield `None` when they are at the latest?